### ISSUE TITLE
Skynet-IADS and ActMobile SAM Sites

### ISSUE THREAD

**Issue by Whiskey-11** [dcs-retribution/dcs-retribution#205]
### Is your feature request related to a problem? Please describe.

Not related to any problems, however an experimental branch of Skynet-IADS has added the ability for Tactical SAM's (SA-6, SA-8, SA-9, SA-11, SA-13, SA-15, SA-17, and SA-19) to use shoot and scoot tactics.  Basically, they'll emit, or fire, then they'll move from that position to a new one where they'll set up and repeat the process.  

### Describe the solution you'd like

Here is a link to the Skynet-IADS ActMobile branch:
https://github.com/baleBaron/Skynet-IADS/tree/baron-branch

Just having Retribution replace the Skynet-Compiled.LUA is not enough to actually cause them to go mobile, but the Skynet-iads-Compiled.LUA from the ActMobile branch is required.

An additional few lines of code are required in the skynetiads-config.lua file Liberation/Retribution spits out when Skynet is used.

At the bottom of the lua file is this section: 
```
    ------------------------------------------------------------------------------------------------------------------------------------------------------------
    -- create the IADS networks
    -------------------------------------------------------------------------------------------------------------------------------------------------------------
    if createRedIADS then
        env.info("DCSLiberation|Skynet-IADS plugin - creating red IADS")
        local redIADS = SkynetIADS:create("IADS")
        initializeIADS(redIADS, 1, includeRedInRadio, debugRED) -- RED      
    end

    if createBlueIADS then
        env.info("DCSLiberation|Skynet-IADS plugin - creating blue IADS")
        local blueIADS = SkynetIADS:create("IADS")
        initializeIADS(blueIADS, 2, includeBlueInRadio, debugBLUE) -- BLUE
    end
```

By changing the RedIADS function you can make specific SAM's mobile:

```
    ------------------------------------------------------------------------------------------------------------------------------------------------------------
    -- create the IADS networks
    -------------------------------------------------------------------------------------------------------------------------------------------------------------
    if createRedIADS then
        env.info("DCSLiberation|Skynet-IADS plugin - creating red IADS")
        local redIADS = SkynetIADS:create("IADS")
        initializeIADS(redIADS, 1, includeRedInRadio, debugRED) -- RED
        redIADS:getSAMSitesByNatoName("SA-11"):setActMobile(true,60*1.75,500,3000,nil)   --ActMobile
        redIADS:getSAMSitesByNatoName("SA-15"):setActMobile(true,60*1.25,250,2000,nil)   --ActMobile
        redIADS:getSAMSitesByNatoName("SA-17"):setActMobile(true,60*1.75,500,3000,nil)   --ActMobile
        redIADS:getSAMSitesByNatoName("SA-19"):setActMobile(true,60*1.25,250,2000,nil)   --ActMobile
    end

    if createBlueIADS then
        env.info("DCSLiberation|Skynet-IADS plugin - creating blue IADS")
        local blueIADS = SkynetIADS:create("IADS")
        initializeIADS(blueIADS, 2, includeBlueInRadio, debugBLUE) -- BLUE
    end
```

The above code sets the SA-11, SA-15, SA-17, and SA-19 to be mobile.  The values after the mobile function control: Is it Mobile, stubbornness (basically how long they'll stay there after an emission, shorter times results in very skiddish SAM's.  The above values are values I've used with success), the minimum distance they'll move, and finally the maximum distance they'll move.  The nil section is where you can define zones they'll travel between, it's not useful for Liberation/Retribution because it requires predefined trigger zones (the names go in the place of nil). To get complete coverage of tactical SAM's, you'd have to add the other tactical SAM's to that list (6, 8, 9, and 13).    

So to recap:  Replacing the skynet-iads-compiled.lua with the ActMobile branch one, and adding those lines to set certain SAM types as mobile, you get Mobile tactical SAM's.  

### Additional context

It would be an awesome addition to the Skynet option in Liberation/Retribution.  Maybe an additional checkbox for ActMobile?  There is no harm (heh!) in using the ActMobile skynet-iads-compiled.lua file for a mission which doesn't have ActMobile enabled as the difference is those lines of code in the skynetiads-config.lua which Liberation/Retribution outputs.

**Comment by Raffson** [comment]
Sounds interesting, I'll try to have a look at this over the weekend if I can find some spare time.

**Comment by Whiskey-11** [comment]
Let me know if there is anything I can do to help.  We were able to do this on our own Liberation campaign by manually replacing the Compiled IADS script and adding those lines in the skynetiads-config.lua file for every liberation mission.  A lot of work, but worth it to make the front line a little more spicy! :)

**Comment by Raffson** [comment]
Got an initial version, although from my tests I noticed that not all SHORADs will be affected by the script.
Specifically independent SHORADs won't be making use of the 'ActMobile' functionality.
I.o.w. if the SHORAD had a PD role, it should work...

**Comment by Whiskey-11** [comment]
Is there a way you can send the changes you made to the skynetiads-config.lua?

If you copied my change exactly, it is only going to add the 11, 15, 17, and 19 to the act mobile list.  That's my bad, I should have included them all.  

```
    ------------------------------------------------------------------------------------------------------------------------------------------------------------
    -- create the IADS networks
    -------------------------------------------------------------------------------------------------------------------------------------------------------------
    if createRedIADS then
        env.info("DCSLiberation|Skynet-IADS plugin - creating red IADS")
        local redIADS = SkynetIADS:create("IADS")
        initializeIADS(redIADS, 1, includeRedInRadio, debugRED) -- RED
        redIADS:getSAMSitesByNatoName("SA-6"):setActMobile(true,60*1.25,250,3000,nil)   --ActMobile
        redIADS:getSAMSitesByNatoName("SA-8"):setActMobile(true,60*1.25,250,3000,nil)   --ActMobile
        redIADS:getSAMSitesByNatoName("SA-9"):setActMobile(true,60*1.25,250,3000,nil)   --ActMobile
        redIADS:getSAMSitesByNatoName("SA-11"):setActMobile(true,60*1.75,500,3000,nil)   --ActMobile
        redIADS:getSAMSitesByNatoName("SA-13"):setActMobile(true,60*1.25,250,3000,nil)   --ActMobile
        redIADS:getSAMSitesByNatoName("SA-15"):setActMobile(true,60*1.25,250,2000,nil)   --ActMobile
        redIADS:getSAMSitesByNatoName("SA-17"):setActMobile(true,60*1.75,500,3000,nil)   --ActMobile
        redIADS:getSAMSitesByNatoName("SA-19"):setActMobile(true,60*1.25,250,2000,nil)   --ActMobile
    end

    if createBlueIADS then
        env.info("DCSLiberation|Skynet-IADS plugin - creating blue IADS")
        local blueIADS = SkynetIADS:create("IADS")
        initializeIADS(blueIADS, 2, includeBlueInRadio, debugBLUE) -- BLUE
    end
```

The above should add all of the REDFOR ones available in DCS + High Digit SAMs.  I have NOT played around with BLUFOR SAM's being added to this list.

**Comment by Raffson** [comment]
Is this what you're looking for?

https://github.com/dcs-retribution/dcs-retribution/commit/2c6c2e5fbde994ca106d3e098d7eb74d85a94134

**Comment by Whiskey-11** [comment]
It looks like everything is there.  I tried downloading the beta branch and getting it to work but I'm borderline stupid with some of these things! :P  

Is there anyway to send the demo miz?  (I'm on the Discord too)

The one thing I know is Skynet LUA shenanigans.  That'd help me diagnose why some of the other SAM's aren't moving.  So long as they are being called up correctly, there is no reason they shouldn't be mobile if the game allows them to be.  The Demo Mission from Baron on the ActMobile Skynet Branch has SA-6, and SA-11, neither of which can do point defense.
